docs/src/main/asciidoc/datasource.adoc

```diff
@@ -155,29 +160,27 @@ quarkus.datasource.password=<your password>
 +
 You can also retrieve the password from Vault by link:{vault-datasource-guide}[using a credential provider] for your datasource.
 
-Until now, the configuration has been the same regardless of whether you’re using a JDBC or a reactive driver.
+Until now, the configuration has been the same regardless of whether you are using a JDBC or a reactive driver.
 When you have defined the database kind and the credentials, the rest depends on what type of driver you are using.
 It is possible to use JDBC and a reactive driver simultaneously.
 
 ==== JDBC datasource
 
 JDBC is the most common database connection pattern, typically needed when used in combination with non-reactive Hibernate ORM.
 
-. Add the `quarkus-agroal` dependency to your project:
+Using a built-in JDBC driver extension automatically includes the Agroal extension, which is the JDBC connection pool implementation applicable for custom and built-in JDBC drivers.
+However, for custom drivers, Agroal needs to be added explicitly.
+
```

Comment:
IMO you could skip explaining this, or move the explanation to the list item where you explain what to do with a custom JDBC driver.

Most people will simply use built-in extensions and won't care about this (and may even get confused).

Reply:
Awesome catch. A did a little shift there. Remove the unnecessary command and replace it with the example + moved the explanation to a NOTE directly above the Agroal step. Makes great sense now (I hope :) )